test/hotspot/gtest/gc/g1/test_freeRegionList.cpp

```diff
@@ -44,7 +44,10 @@ TEST_OTHER_VM(G1FreeRegionList, length) {
 
   // Create a fake heap. It does not need to be valid, as the G1HeapRegion constructor
   // does not access it.
-  MemRegion heap(nullptr, num_regions_in_test * G1HeapRegion::GrainWords);
+  HeapWord val{};
+  HeapWord* ptr = align_up(&val, os::vm_page_size());
+
+  MemRegion heap(ptr, num_regions_in_test * G1HeapRegion::GrainWords);
```

Comment:
I would not do it this way. The MemRegion is larger than your stack, and your stack may border on other stacks, and this test accidentally writing into the region if someone messes the test up could lead to weird crashes sometime later. Its easier to just crash right on accessing uncommitted memory:

```suggestion
  const size_t szw = num_regions_in_test * G1HeapRegion::GrainWords;
  const size_t sz = szw * BytePerWord;
  char* addr = os::reserve_memory(sz, mtTest);
  MemRegion heap((HeapWord*)ptr, szw);
```
and then after the test release it: `os::release_memory(addr, szw);`